android/src/main/java/com/rt2zz/reactnativecontacts/ContactsManager.java

```diff
@@ -605,23 +657,26 @@ public void updateContact(ReadableMap contact, Callback callback) {
             ops.add(op.build());
         }
 
-        // remove existing emails first
-        op = ContentProviderOperation.newDelete(ContactsContract.Data.CONTENT_URI)
-                .withSelection(
-                    ContactsContract.Data.MIMETYPE  + "=? AND "+ ContactsContract.Data.CONTACT_ID + " = ?", 
-                    new String[]{String.valueOf(CommonDataKinds.Email.CONTENT_ITEM_TYPE), String.valueOf(recordID)}
-                );
-        ops.add(op.build());
-
-        // add passed email addresses
-        for (int i = 0; i < numOfEmails; i++) {
-            op = ContentProviderOperation.newInsert(ContactsContract.Data.CONTENT_URI)
-                    .withValue(ContactsContract.Data.RAW_CONTACT_ID, String.valueOf(rawContactId))
-                    .withValue(ContactsContract.Data.MIMETYPE, CommonDataKinds.Email.CONTENT_ITEM_TYPE)
-                    .withValue(CommonDataKinds.Email.ADDRESS, emails[i])
-                    .withValue(CommonDataKinds.Email.TYPE, emailsLabels[i]);
-            ops.add(op.build());
-        }
+		if (emailAddresses != null){
+			// remove existing emails first
+			op = ContentProviderOperation.newDelete(ContactsContract.Data.CONTENT_URI)
+					.withSelection(
+						ContactsContract.Data.MIMETYPE  + "=? AND "+ ContactsContract.Data.RAW_CONTACT_ID + " = ?", 
+						new String[]{String.valueOf(CommonDataKinds.Email.CONTENT_ITEM_TYPE), String.valueOf(rawContactId)}
+					);
+			ops.add(op.build());
+
+			// add passed email addresses
+			for (int i = 0; i < numOfEmails; i++) {
+				op = ContentProviderOperation.newInsert(ContactsContract.Data.CONTENT_URI)
+						.withValue(ContactsContract.Data.RAW_CONTACT_ID, String.valueOf(rawContactId))
+						.withValue(ContactsContract.Data.MIMETYPE, CommonDataKinds.Email.CONTENT_ITEM_TYPE)
+						.withValue(CommonDataKinds.Email.ADDRESS, emails[i])
+						.withValue(CommonDataKinds.Email.TYPE, emailsTypes[i])
```

Comment:
What happens if `TYPE` == `TYPE_WORK`? Since the label is then already defined by the TYPE, essentially `.withValue(CommonDataKinds.Email.LABEL, emailsLabels[i]);` becomes redundant right? How does android behave in this case, will it add double labels?

Reply:
I also have a few other thoughts about how default labels work as well. 
When I started working on it, it took match all lowercase inputs to the default labels, and I kept it that way.  (For example: `{label: work}` will give you `{TYPE: TYPE_WORK}`, and `{label: "Work"}` will give you `{TYPE: TYPE_CUSTOM, LABEL: "Work"}`)
I'm not sure whether converting the input label to all lowercase for type matching should be on the user's end or react-native-contact's end.